### PR TITLE
Builds facility dictionary

### LINKED ISSUE #1: Parse Cyclus input file
XML

### PR DESCRIPTION
**Pull request by nuclearkatie**
Supersedes #5, attempts to incorporate all feedback

Closes #1

### CODE REVIEW

**Comment by nuclearkatie** [comment]
Needs tests

File: tests/test_parse_input.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+def test_find_commod():
+    tree = ET.parse('input/source_1_sink_1.xml')
+    root = tree.getroot()
+    out_commod_tags = ['commodity']
+    for archetype_var in root.find('./facility/config/Source'):
+        commods = pi.find_commod(archetype_var, 'outcommod')
+
+        assert out_commod_tags == commods
+        break
```
Comment (bam241) [inline tests/test_parse_input.py:14]: why break here, I feel like it will interrupt your `for` loop

File: trailmap/parse_input.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import cyclus
+
+
+def parse_input(input, commodity_dictionary):
+    '''Builds dicionary of facilities and their incommodies and outcommodities.
+
+    inputs:
+        - input: a Cyclus XML input file
+        - commodity_dictionary: a dictionary
+
```
Comment (bam241) [inline trailmap/parse_input.py:15]: I would remove that blank line here. (and all the others in the docs strings.)
r at least make it consistent with `get_facility_and_commod_names` methods

File: trailmap/parse_input.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    facility_dict_out = {}
+
+    for facility in root.findall('./facility'):
+        facility_name = facility.find('name').text
+        facility_archetype = facility.find('config/').tag
+        facility_module = input_archetypes[facility_archetype]
+        commodities = commodity_dictionary[facility_module]
+
+        (in_commod_tags,
+         out_commod_tags) = commodity_dictionary[facility_module]
```
Comment (bam241) [inline trailmap/parse_input.py:58]: I would break the line at the `=` sign not in between the pair

File: trailmap/parse_input.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    if archetype_tag.tag in commod_tags:
+        vals = archetype_tag.findall('./')
+        if vals:
+            for commod in vals:
+                commod_list.append(commod.text)
+        else:
+            commod_list.append(archetype_tag.text)
+        return commod_list
+
+    return
```
Comment (bam241) [inline trailmap/parse_input.py:125]: maybe you can always `return commod_list`
empty or filled

File: trailmap/parse_input.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        {'facility' : (['incommodities'], ['outcommodities'])}
+    '''
+    facility_dict_in = {}
+    facility_dict_out = {}
+
+    for facility in root.findall('./facility'):
+        facility_name = facility.find('name').text
+        facility_archetype = facility.find('config/').tag
+        facility_module = input_archetypes[facility_archetype]
+        commodities = commodity_dictionary[facility_module]
```
Comment (bam241) [inline trailmap/parse_input.py:55]: Maybe you  should use `commodities[0]` and `commodities[1]` instead multiplying variables

File: trailmap/parse_input.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+        (in_commod_tags,
+         out_commod_tags) = commodity_dictionary[facility_module]
+
+        facility_in_commods = []
+        facility_out_commods = []
+
+        for archetype_var in facility.find('.config/' + facility_archetype):
+
```
Comment (bam241) [inline trailmap/parse_input.py:64]: maybe that blank line is not necessary ?

File: trailmap/parse_input.py (hunk 5 above)
Comment (bam241) [reply trailmap/parse_input.py:55]: (you are not using `commodities`)
you use `in_commod_tags` and `out_commod_tags` instead

Comment (nuclearkatie) [reply trailmap/parse_input.py:55]: Thank you, forgot to take that out

File: tests/test_parse_input.py (hunk 1 above)
Comment (nuclearkatie) [reply tests/test_parse_input.py]: fixed so it doesn't just look at the first pass through the loop. Thanks for pointing that out!

**Comment by gonuke** [comment]
It's helpful to name variables in tests with prefixes `exp_` for the expected results and `obs_` for the observed results.

**Comment by gonuke** [comment]
Looks good - thanks @nuclearkatie

**Pull request merged**